### PR TITLE
Bug fix/bts 68

### PR DESCRIPTION
**Pull request by graetzer**
close when done: https://arangodb.atlassian.net/projects/BTS/issues/BTS-68

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10133/

docs PR: https://github.com/arangodb/docs/pull/464

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10145/

File: arangod/Pregel/GraphStore.cpp [hunk 1]
```diff
@@ -438,7 +438,12 @@ void GraphStore<V, E>::_loadEdges(transaction::Methods& trx, Vertex<V, E>& verte
       TRI_ASSERT(edgeSlice.isString());
       
       VPackStringRef toValue(edgeSlice);
-      allocateSpace(toValue.size());
+      size_t space = toValue.size();
+      
+      
+//      if constexpr
+      
+      allocateSpace(space);
```
Comment (jsteemann) [inline arangod/Pregel/GraphStore.cpp:446]:
```suggestion
      size_t space = toValue.size();
      allocateSpace(space);
```

File: arangod/Pregel/Graph.h [hunk 2]
```diff
@@ -48,6 +48,10 @@ struct PregelID {
   bool operator==(const PregelID& rhs) const {
     return shard == rhs.shard && key == rhs.key;
   }
+  
+  bool operator!=(const PregelID& rhs) const {
+    return shard != rhs.shard || key != rhs.key;
```
Comment (jsteemann) [inline arangod/Pregel/Graph.h:53]: this is normally implemented as the negation of the `==` operator, but it is ok as it is:
```suggestion
    return !(*this == rhs);
```

File: arangod/Pregel/Algos/WCC.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+struct WCCGraphFormat final : public GraphFormat<uint64_t, uint64_t> {
+  explicit WCCGraphFormat(application_features::ApplicationServer& server,
+                         std::string const& result)
+      : GraphFormat<uint64_t, uint64_t>(server), _resultField(result) {}
+  
+  std::string const _resultField;
+  
+  size_t estimatedVertexSize() const override { return sizeof(uint64_t); };
+  size_t estimatedEdgeSize() const override { return sizeof(uint64_t); };
```
Comment (jsteemann) [inline arangod/Pregel/Algos/WCC.cpp:103]:
```suggestion
  size_t estimatedVertexSize() const override { return sizeof(uint64_t); }
  size_t estimatedEdgeSize() const override { return sizeof(uint64_t); }
```

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10235/

**Comment by jsteemann** [comment]
Pulled in current devel and restarted Jenkins:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10238/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10369/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10374/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**